cpp/arcticdb/python/python_to_tensor_frame.cpp

```diff
@@ -68,19 +68,34 @@ NativeTensor obj_to_tensor(PyObject *ptr) {
             auto none = py::none{};
             auto obj = reinterpret_cast<PyObject **>(arr->data);
             bool empty = false;
+            bool all_nans = false;
             PyObject *sample = *obj;
             if (sample == none.ptr() || is_py_nan(sample)) {
-                // Iterate till we find the first non null element, the frontend ensures there is at least one.
+                empty = true;
+                all_nans = true;
                 util::check(c_style, "Non contiguous columns with first element as None not supported yet.");
-                auto casted_col =
-                    std::find_if(obj, obj + size, [&none](auto val) { return val != none.ptr() && !is_py_nan(val); });
-                empty = casted_col == obj + size;
-                sample = *casted_col;
+                PyObject** current_object = obj;
```

Comment:
```suggestion
                // Iterating on the content of the NumPy array to verify if all
                // elements are NaNs or if the NumPy array is empty.
                PyObject** current_object = obj;
```

Reply:
I'm personally not a fan of such type of comments. There is not sophisticated algorithm. If anyone else believes it adds value I'll add it.

Reply:
I've added a comment about the way we treat NaN and None, which also explains why we need this loop